lib/plausible_web/controllers/site_controller.ex

```diff
@@ -867,4 +872,53 @@ defmodule PlausibleWeb.SiteController do
         |> redirect(to: Routes.site_path(conn, :settings_general, site.domain))
     end
   end
+
+  def change_domain(conn, _params) do
+    if Plausible.v2?() do
+      site = conn.assigns[:site]
+
+      changeset = Plausible.Site.update_changeset(site)
```

Comment:
Nitpick, feel free to ignore:

```suggestion
      changeset = Plausible.Site.update_changeset(conn.assigns.site)
```